reconcile/aus/base.py

```diff
@@ -144,23 +261,23 @@ def fetch_desired_state(
                 policy = copy.deepcopy(upgrade_policy)
                 policy["addon_id"] = addon["id"]
                 policy["current_version"] = addon["version"]
-                desired_state.append(policy)
+
+                desired_state.append(ConfiguredAddonUpgradePolicy(**policy))
         else:
             spec = ocm.clusters[cluster_name].spec
             upgrade_policy["current_version"] = spec.version
             upgrade_policy["channel"] = spec.channel
             upgrade_policy["available_upgrades"] = ocm.available_cluster_upgrades.get(
                 cluster_name
             )
-            desired_state.append(upgrade_policy)
+            desired_state.append(ConfiguredClusterUpgradePolicy(**upgrade_policy))
 
-    sorted_desired_state = sorted(desired_state, key=sort_key)
-
-    return sorted_desired_state
+    return sorted(desired_state, key=sort_key)
 
 
+# This does not look like it belongs here, perhaps cluster_version_data is more
```

Comment:
i would love to know the end of this sentence

Reply:
Thought about moving this into `reconcile/utils/cluster_version_data.py`
Perhaps worth considering for a follow-up refactoring. Additionally looking at the test code, I think it could need some restructuring as well.